projects/core/src/occ/adapters/product/converters/product-image-normalizer.ts

```diff
@@ -55,14 +55,14 @@ export class ProductImageNormalizer implements Converter<Occ.Product, Product> {
          * `backend.media.baseUrl` by default, but fallback to `backend.occ.baseUrl`
          * if none provided.
          */
-        targetImage.url =
-          (this.config.backend.media.baseUrl ||
-            this.config.backend.occ.baseUrl ||
-            '') + image.url;
-
+        targetImage.url = this.processTargetImageUrl(targetImage.url);
         imageContainer[image.format] = targetImage;
       }
     }
     return images;
   }
+
+  private processTargetImageUrl(url: string){
+    return  url.startsWith('http') ? url : (this.config.backend.media.baseUrl || this.config.backend.occ.baseUrl || '') + url;
```

Comment:
I'd prepare for more cases here:
- uppercase / lowercase/ mixed case
- the `//` syntax (allows for cross http / https)

We could even consider other protocols or the `data:image` for data URLs, you could evaluate an array of prefixes and test against it (_contains_) Given that we're concerned with the OCC backend, those are really edges cases though. Up to you!